lib/index.d.ts

```diff
@@ -550,31 +549,27 @@ declare namespace planck {
         getSupportVertex(d: Vec2): Vec2;
         set(shape: Shape, index: number): void;//TODO index is only used by Chain
     }
-    type Shape = CircleShape | EdgeShape | PolygonShape | ChainShape;
+    interface Shape {
```

Comment:
This destroys my hope of getting implict typecasting working 😞 , but I admit it got way too verbose 😄

Reply:
Can you please explain it a little more, I'm new to TS.

Reply:
In https://github.com/shakiba/planck.js/pull/61 I explained currently
```
if (a.m_type === "circle") {
    a // type: planck.CircleShape
}
```
works (so you don't have to cast `a` to `CircleShape`. (It currently does not work with `a.getType()` but I think thats a bug of TS). But this trick only works if `Shape` is defined as union of the derived classes, cause otherwise there could be another class extending `Shape` with `m_type` "circle". TS is powerful enough to interfer the methods that all derived classes have in common and make them callable on the general `Shape` type.

But a you noticed it's quite verbose and I don't know if its worth. 🤔 Maybe change it as you did for now; if some day it works with `.getType()` we can discuss it then in an separate issue.